### DISCUSSION TITLE
Tray Icon badge no longer working

### DISCUSSION THREAD

**Discussion by mspronesti** [cstayyab/WALC#94] [Troubleshooting]
After upgrading to latest release tray icon doesn't show notification badge any longer nor gets grey when connection is missing. I tried reverting to previous version and it doesn't work either.

**Comment by cstayyab** [comment]
It must be the issues related to new WhatsApp Web as suggested in this comment https://github.com/cstayyab/WALC/issues/90#issuecomment-749656463

**Comment by mspronesti** [comment]
Will you guys fix it in the next release ?

**Comment by daffamumtaz2361** [comment]
This issue is most likely due to outdated whatsap-web.js, updating it should fix the issue. It already happened in the past.

I'm actually planning to do some major rework with the app, just need to coordinate with @cstayyab first.

**Reply by cstayyab** [reply]
Major rework would be good. I have somethings in my mind too about major rework. You can communicate your plan and we can create a project here in this repository so tracking progress is easy.

**Reply by daffamumtaz2361** [reply]
@cstayyab I already sent you an email about it. I have some questions there, please answer it.

**Reply by bkmo** [reply]
Is this on a to-do list?

**Reply by daffamumtaz2361** [reply]
Yes, I'm working on it on my own fork. It already worked, but then whatsapp has to ruin it :( an update caused a breakage. Great, back where we started.

**Reply by bkmo** [reply]
That figures......I did have a new whatsap update not long ago.

**Reply by hiddenman** [reply]
Hi,

Thank you, guys, for working on this. If you any help, just tell us.  We all need this badges, otherwise the trayicon is useless.

**Reply by hiddenman** [reply]
> Yes, I'm working on it on my own fork. It already worked, but then whatsapp has to ruin it :( an update caused a breakage. Great, back where we started.

@Zzombiee2361 I see you already fixes all the issues in your fork. Are you going to pull-request them or may be you will build it by yourself?
Thank you.

**Reply by hiddenman** [reply]
> > Yes, I'm working on it on my own fork. It already worked, but then whatsapp has to ruin it :( an update caused a breakage. Great, back where we started.
> 
> @Zzombiee2361 I see you already fixes all the issues in your fork. Are you going to pull-request them or may be you will build it by yourself?
> Thank you.

@Zzombiee2361 Just tried to run your version and got a problem with the patch:

```
patch-package 6.4.6
Applying patches...

**ERROR** Failed to apply patch for package whatsapp-web.js at path
  
    node_modules/whatsapp-web.js

  This error was caused because patch-package cannot apply the following patch file:

    patches/whatsapp-web.js+1.12.5.patch

  Try removing node_modules and trying again. If that doesn't work, maybe there was
  an accidental change made to the patch file? Try recreating it by manually
  editing the appropriate files and running:
  
    patch-package whatsapp-web.js
  
  If that doesn't work, then it's a bug in patch-package, so please submit a bug
  report. Thanks!

    https://github.com/ds300/patch-package/issues
```

**Reply by daffamumtaz2361** [reply]
@hiddenman Ok, I fixed the patch. To install it you need to run `npm install` then when finished, delete the `package-lock.json` and run `npm install` again. This is to apply the changes in the patch. You then can hopefully run `npm start` successfully.

I know this is a mess currently, I'm looking forward to make my own `whatsapp-web.js` fork for electron since @cstayyab fork is outdated and he's been unresponsive. Any idea what to name the fork?

**Reply by hiddenman** [reply]
> @hiddenman Ok, I fixed the patch. To install it you need to run `npm install` then when finished, delete the `package-lock.json` and run `npm install` again. This is to apply the changes in the patch. You then can hopefully run `npm start` successfully.
> 
> I know this is a mess currently, I'm looking forward to make my own `whatsapp-web.js` fork for electron since @cstayyab fork is outdated and he's been unresponsive. Any idea what to name the fork?

Thank you. I just made a fresh clone and did what you'd said, but got it again:

`**ERROR** Failed to apply patch for package whatsapp-web.js at path`

I ran:
npm install
rm rm package-lock.json
npm install

Thank you.

**Reply by hiddenman** [reply]
> > @hiddenman Ok, I fixed the patch. To install it you need to run `npm install` then when finished, delete the `package-lock.json` and run `npm install` again. This is to apply the changes in the patch. You then can hopefully run `npm start` successfully.
> > I know this is a mess currently, I'm looking forward to make my own `whatsapp-web.js` fork for electron since @cstayyab fork is outdated and he's been unresponsive. Any idea what to name the fork?
> 
> Thank you. I just made a fresh clone and did what you'd said, but got it again:
> 
> `**ERROR** Failed to apply patch for package whatsapp-web.js at path`
> 
> I ran:
> npm install
> rm rm package-lock.json
> npm install
> 
> Thank you.

Do you have a fresh version of the appimage with all your updates?

**Comment by daffamumtaz2361** [comment]
This is currently working on my fork but currently the build process is complicated and contains many deal breaking bugs such as #110, I haven't been able to pinpoint the issue. I do have a working AppImage build and if there's enough demand I could share it. Any help is appreciated

**Reply by bkmo** [reply]
@ Zzombiee2361 Your work is appreciated. A fork would be welcome seeing that nothing is being done here. I would give the new appimage a go if you are thinking about sharing.

**Reply by daffamumtaz2361** [reply]
[Here's the AppImage](https://drive.google.com/file/d/1YtgQC4wtOULTb2Qmgs1-PZEhxfXmQzZD/view?usp=sharing), though be wary #110 is really deal breaking for me. The file picker is causing crash, I basically can't download or upload files (dropping it to upload works though). Until it is fixed I can't properly release a new version.

**Reply by bkmo** [reply]
I am running the appimage, and first thing I notice is the click behavior of the tray icon. A left click acts as a right click and brings up the context menu. This could be another issue with electron though.

**Reply by daffamumtaz2361** [reply]
It is, I used to lock the electron version to 8.3.1 because of this issue, see #47 for more info. We have to update eventually, so now it's not working. Bummer, I know. I actually want to integrate it better with KDE so it can display badge in task manager, hide the tray icon when there's no notification, and even use inline reply in notification. Though I don't know where to start integrating C++ code with node.js

**Comment by daffamumtaz2361** [comment]
Ok, I finally fixed the tray icon badge, sent a pr (#125) and merged it. Now we wait for @cstayyab to release the new changes. For those who are impatient or want to test it out before release can download it from my google drive [here](https://drive.google.com/file/d/1GBV2do6xrCWmX6ZS7H0NbtJo55Air7YF/view?usp=sharing). Issue #110 is no longer happening, can anyone confirm it's fixed?

Interesting thing is this file is smaller than previous AppImage, 78 MB vs 84.3 MB. Probably because it's now using my own fork of `whatsapp-web.js` which removes some unnecessary packages like `puppeteer`

**Reply by hiddenman** [reply]
> Ok, I finally fixed the tray icon badge, sent a pr (#125) and merged it. Now we wait for @cstayyab to release the new changes. For those who are impatient or want to test it out before release can download it from my google drive [here](https://drive.google.com/file/d/1GBV2do6xrCWmX6ZS7H0NbtJo55Air7YF/view?usp=sharing). Issue #110 is no longer happening, can anyone confirm it's fixed?
> 
> Interesting thing is this file is smaller than previous AppImage, 78 MB vs 84.3 MB. Probably because it's now using my own fork of `whatsapp-web.js` which removes some unnecessary packages like `puppeteer`

Hi,

Thank you very much!

1. I cloned and ran it successfully.2
2. The tray and task manager icon has been changed to the icon with a penguin inside :)
3. Tray count badge works correctly
4. I can download attachments (tried some).

**Reply by hiddenman** [reply]
> Ok, I finally fixed the tray icon badge, sent a pr (#125) and merged it. Now we wait for @cstayyab to release the new changes. For those who are impatient or want to test it out before release can download it from my google drive [here](https://drive.google.com/file/d/1GBV2do6xrCWmX6ZS7H0NbtJo55Air7YF/view?usp=sharing). Issue #110 is no longer happening, can anyone confirm it's fixed?
> 
> Interesting thing is this file is smaller than previous AppImage, 78 MB vs 84.3 MB. Probably because it's now using my own fork of `whatsapp-web.js` which removes some unnecessary packages like `puppeteer`

@Zzombiee2361 One more comment - now clicking right and left mouse button brings the context menu. In the previous version left-clicking immediately raised it from the tray, it was comfortable. Now i always have to click and then choose Show WALC - too many action. I suppose left-clicking should raise/hide, right-clicking should bring the context menu.

Thank you.

**Reply by bkmo** [reply]
That behavior comes from a decision fom the Electron end of things. The newer Electron version being used now causes this behavior. Kinda sucks doesn't it? I am not sure if this is KDE specific, as I am sure it works correctly in some DE's

**Reply by bkmo** [reply]
This version seems to be working fine. Only issue I have is it occasionally notifies me with a network connection lost on and off. It does this several times even when the network is up solid. How prone is this to break again with a Whatsapp update?

**Reply by hiddenman** [reply]
@Zzombiee2361 Hi,

We've got this issue again, not sure why. Just stopped working again, here is the new issues from other guys: https://github.com/cstayyab/WALC/issues/157
